<launch>

  <!--

  HEBI MoveIt! node config file

  The content of this is largely taken from the
  standard MoveIt launch files, but is configured to
  use the `arm_moveit_node` in the `hebi_cpp_api_examples`
  package to move a physical arm.

  You should specify `arm_type` as a launch file parameter
  (defaults to the HEBI 6-DoF arm a-2085-06).  This should
  match a moveit config that is visible as a package with
  name `hebi_<arm_type>_moveit_config`, as well as a config
  parameters file located at `config/<arm_type>_params.yaml`.

  -->

  <!-- specify type of HEBI arm -->
  <arg name="arm_type" default="a-2085-06" />

  <!-- specify gripper type (e.g., 'parallel' or 'none') -->
  <arg name="gripper_type" default="none" />

  <!-- The moveit configuration matching the above args -->
  <arg name="moveit_arm_type" value="$(eval (arm_type + '-parallel-gripper') if gripper_type == 'parallel' else arm_type)"/> 

  <!-- specify the planning pipeline -->
  <arg name="pipeline" default="ompl" />

  <!-- By default, we are not in debug mode -->
  <arg name="debug" default="false" />

  <!-- Parameters for the module names, gains, hrdf, etc -->
  <rosparam file="$(find hebi_cpp_api_examples)/config/$(arg arm_type)_params.yaml" command="load" />

  <node pkg="hebi_cpp_api_examples" type="arm_moveit_node" name="arm_moveit_node" output="screen"/>

  <!-- Start gripper node as appropriate -->
  <group if="$(eval gripper_type == 'parallel')">
    <!-- TODO -->
    <!-- <node pkg="hebi_cpp_api_examples" type="moveit_gripper_node" name="moveit_gripper_node" output="screen"/> -->
  </group>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />

  <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <include file="$(eval find('hebi_' + moveit_arm_type + '_moveit_config') + '/launch/planning_context.launch')">
    <arg name="load_robot_description" value="true"/>
  </include>

  <!-- Run the main MoveIt! executable -->
  <include file="$(eval find('hebi_' + moveit_arm_type + '_moveit_config') + '/launch/move_group.launch')">
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="pipeline" value="$(arg pipeline)"/>
  </include>

  <!-- Run Rviz and load the default config to see the state of the move_group node -->
  <include file="$(eval find('hebi_' + moveit_arm_type + '_moveit_config') + '/launch/moveit_rviz.launch')">
    <arg name="rviz_config" value="$(eval find('hebi_' + moveit_arm_type + '_moveit_config') + '/launch/moveit.rviz')"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>

</launch>
